File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/base/dto/FileDto.java
package cc.nobrain.dev.userserver.domain.base.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;

/**
 * DTO for {@link cc.nobrain.dev.userserver.domain.base.entity.File}
 */
@Data
@Accessors(chain = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class FileDto extends BaseDto implements Serializable {
    private Long id;
    private String dtype;
    private String url;
    @NotNull
    private String fileName;
    private String originalFileName;
    @NotNull
    private Long size;
    private String mimeType;
}

File: UserServer/src/main/kotlin/cc/nobrain/dev/userserver/domain/base/dto/FileDto.kt
package cc.nobrain.dev.userserver.domain.base.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.validation.constraints.NotNull
import java.io.Serializable

@JsonIgnoreProperties(ignoreUnknown = true)
open class FileDto(
    var id: Long? = null,
    var dtype: String? = null,
    var url: String? = null,
    @field:NotNull
    var fileName: String? = null,
    var originalFileName: String? = null,
    @field:NotNull
    var size: Long? = null,
    var contentType: String? = null
) : BaseDto(), Serializable
